okhttp/src/main/java/okhttp3/OkHttpClient.kt

```diff
@@ -0,0 +1,829 @@
+      this.proxyAuthenticator = okHttpClient.proxyAuthenticator
+      this.authenticator = okHttpClient.authenticator
+      this.connectionPool = okHttpClient.connectionPool
+      this.dns = okHttpClient.dns
+      this.followSslRedirects = okHttpClient.followSslRedirects
+      this.followRedirects = okHttpClient.followRedirects
+      this.retryOnConnectionFailure = okHttpClient.retryOnConnectionFailure
+      this.callTimeout = okHttpClient.callTimeout
+      this.connectTimeout = okHttpClient.connectTimeout
+      this.readTimeout = okHttpClient.readTimeout
+      this.writeTimeout = okHttpClient.writeTimeout
+      this.pingInterval = okHttpClient.pingInterval
+    }
+
+    /**
+     * Sets the default timeout for complete calls. A value of 0 means no timeout, otherwise values
+     * must be between 1 and [Integer.MAX_VALUE] when converted to milliseconds.
+     *
+     * The call timeout spans the entire call: resolving DNS, connecting, writing the request
+     * body, server processing, and reading the response body. If the call requires redirects or
+     * retries all must complete within one timeout period.
+     *
+     * The default value is 0 which imposes no timeout.
+     */
+    fun callTimeout(timeout: Long, unit: TimeUnit) = apply {
+      callTimeout = checkDuration("timeout", timeout, unit)
+    }
+
+    /**
+     * Sets the default timeout for complete calls. A value of 0 means no timeout, otherwise values
+     * must be between 1 and [Integer.MAX_VALUE] when converted to milliseconds.
+     *
+     * The call timeout spans the entire call: resolving DNS, connecting, writing the request
+     * body, server processing, and reading the response body. If the call requires redirects or
+     * retries all must complete within one timeout period.
+     *
+     * The default value is 0 which imposes no timeout.
+     */
+    @IgnoreJRERequirement
+    fun callTimeout(duration: Duration) = apply {
+      callTimeout = checkDuration("timeout", duration.toMillis(), TimeUnit.MILLISECONDS)
+    }
+
+    /**
+     * Sets the default connect timeout for new connections. A value of 0 means no timeout,
+     * otherwise values must be between 1 and [Integer.MAX_VALUE] when converted to
+     * milliseconds.
+     *
+     * The connect timeout is applied when connecting a TCP socket to the target host.
+     * The default value is 10 seconds.
+     */
+    fun connectTimeout(timeout: Long, unit: TimeUnit) = apply {
+      connectTimeout = checkDuration("timeout", timeout, unit)
+    }
+
+    /**
+     * Sets the default connect timeout for new connections. A value of 0 means no timeout,
+     * otherwise values must be between 1 and [Integer.MAX_VALUE] when converted to
+     * milliseconds.
+     *
+     * The connect timeout is applied when connecting a TCP socket to the target host.
+     * The default value is 10 seconds.
+     */
+    @IgnoreJRERequirement
+    fun connectTimeout(duration: Duration) = apply {
```

Comment:
apply is a nice trick

Reply:
That is great for the builder pattern, nice find!